glotaran/builtin/models/kinetic_image/kinetic_image_dataset_descriptor.py

```diff
@@ -12,11 +12,38 @@
     }
 )
 class KineticImageDatasetDescriptor(DatasetDescriptor):
-    def get_k_matrices(self):
-        return [mat for mat in [cmplx.full_k_matrix() for cmplx in self.megacomplex] if mat]
+    def get_megacomplex_k_matrices(self):
+        scales = (
+            [
+                self.megacomplex_scale[i]
+                for i in [
+                    i
+                    for i, megacomplex in enumerate(self.megacomplex)
+                    if megacomplex.has_k_matrix()
+                ]
+            ]
+            if self.megacomplex_scale is not None
+            else None
+        )
+
+        matrices = [
+            matrix
+            for matrix in [
+                megacomplex.full_k_matrix()
+                for megacomplex in self.megacomplex
+                if megacomplex.has_k_matrix()
+            ]
+        ]
+
+        return scales, matrices
+
+    def has_k_matrix(self):
+        _, matrices = self.get_megacomplex_k_matrices()
+        return len(matrices) != 0
```

Comment:
```suggestion
    def has_k_matrix(self) -> bool:
        _, k_matrices = self.get_megacomplex_k_matrices()
        return len(k_matrices) != 0
```
Just for concistency.